### PR TITLE
[dxva] Fixed: don't use DXVADDI_Intel_ModeH264_E decoder if source re…

### PR DESCRIPTION
**Pull request by afedchin**
…quires more than 11 references.

this is backport of #7647

### CODE REVIEW

**Comment by fritsch** [comment]
@afedchin is this a windows limitation? On Linux we nicely decode H264 Level5.1@16 Reframes?

**Comment by afedchin** [comment]
@fritsch I don't know exactly how hw decoding works on linux. Perhaps this is dxva only issue.

**Comment by fritsch** [comment]
So - this won't work: https://dl.dropboxusercontent.com/u/55728161/clannad-5.1.mkv anymore on windows then?

**Comment by afedchin** [comment]
It will be decoded by sw on legacy Intel GPUs like Intel Arrandale and older.

**Comment by afedchin** [comment]
Sandy and Ivy has generic h264 (H.264 VLD, no FGT) decoder which works fine with resf > 11.

**Comment by fritsch** [comment]
Okay - that makes it clear - fine with me.

File: xbmc/cores/dvdplayer/DVDCodecs/Video/DXVA.cpp [hunk 1]
```diff
@@ -818,7 +818,15 @@ bool CDecoder::Open(AVCodecContext *avctx, AVCodecContext* mainctx, enum PixelFo
     return false;
   }
 
-  m_format.SampleWidth  = avctx->coded_width;
+  CLog::Log(LOGDEBUG, "DXVA - source requires %d references", avctx->refs);
+  if (m_input == DXVADDI_Intel_ModeH264_E && avctx->refs > 11)
+  {
+    const dxva2_mode_t *mode = dxva2_find_mode(&m_input);
+    CLog::Log(LOGWARNING, "DXVA - too many refelences %d for selected decoder '%s'.", avctx->refs, mode->name);
```
Comment (phil65) [inline xbmc/cores/dvdplayer/DVDCodecs/Video/DXVA.cpp]: small typo here ("refelences" vs. "references")

Comment (afedchin) [inline xbmc/cores/dvdplayer/DVDCodecs/Video/DXVA.cpp]: thanks. I'll fix.

**Comment by afedchin** [comment]
jenkins build this please

**Pull request merged**